src/DynamoCoreWpf/Views/Preview/WatchTree.xaml.cs

```diff
@@ -39,18 +40,53 @@ private void Button_Click(object sender, RoutedEventArgs e)
                 node.Click();
         }
 
-        private void TreeView1_OnSelectedItemChanged(object sender, RoutedPropertyChangedEventArgs<object> e)
+        private void treeviewItem_MouseLeftButtonUp(object sender, System.Windows.Input.MouseButtonEventArgs e)
         {
-            var node = e.NewValue as WatchViewModel;
-            if (node == null)
+            TreeViewItem tvi = (TreeViewItem)sender;
+            var node = tvi.DataContext as WatchViewModel;
+
+            HandleItemChanged(tvi, node);
+
+            e.Handled = true; 
+        }
+        private void treeviewItem_KeyUp(object sender, System.Windows.Input.KeyEventArgs e)
+        {
+            if (e.Key == System.Windows.Input.Key.Up || e.Key == System.Windows.Input.Key.Down)
```

Comment:
Yes, previously, it was controlled by the `SelectedItemChanged` Event. This, however, does not allow for 'deselection' of the node. As such, I had to implement a workaround by going to lower level Events such as `Mouse Up` and `KeyUp`. I searched the web and thought about how else to implement it, but it seems that this is an inherent limitation in the TreeView framework and hence, the workaround.